pkg/receive/handler.go

```diff
@@ -245,17 +313,37 @@ func (h *Handler) forward(ctx context.Context, tenant string, wreq *prompb.Write
 	// to every other node in the hashring, rather than
 	// one request per time series.
 	for i := range wreq.Timeseries {
-		endpoint, err := h.hashring.Get(tenant, &wreq.Timeseries[i])
+		endpoint, err := h.hashring.GetN(tenant, &wreq.Timeseries[i], replica)
 		if err != nil {
 			return nil, err
 		}
 		if _, ok := wreqs[endpoint]; !ok {
 			wreqs[endpoint] = &prompb.WriteRequest{}
+			replicas[endpoint] = replica
 		}
 		wr := wreqs[endpoint]
 		wr.Timeseries = append(wr.Timeseries, wreq.Timeseries[i])
 	}
 
+	var errs error
+	// Collect all the errors into a single error.
+	for _, err := range h.parallelizeRequests(ctx, tenant, replicas, wreqs, forwardRequestsTotal, forwardErrorsTotal) {
+		if err != nil {
+			if errs == nil {
+				errs = err
+				continue
+			}
+			errs = errors.Wrap(errs, err.Error())
```

Comment:
MultiErrror makes more sense here no? In fact `.parallelizeRequsets` could immediately return a MultiError

Reply:
Sounds good! I didn’t realize we had a util for that :)